Backend/WebApi-PointMap/Controllers/UserManagementController.cs

```diff
@@ -198,140 +169,111 @@ public IHttpActionResult DeleteUser([FromBody, Required] LoginDTO requestPayload
             }
-            Guid userSSOID;
             try
             {
-                // check if valid SSO ID format
-                userSSOID = Guid.Parse(requestPayload.SSOUserId);
+                var userSSOID = ParseAndCheckId(requestPayload.SSOUserId);
+
                 // check valid signature
                 TokenService _tokenService = new TokenService();
                 if (!_tokenService.isValidSignature(requestPayload.PreSignatureString(), requestPayload.Signature))
                 {
                     throw new InvalidTokenSignatureException("Session is not valid.");
                 }
-            }
-            catch (InvalidTokenSignatureException ex)
-            {
-                return Content((HttpStatusCode)401, ex.Message);
-            }
-            catch (Exception)
-            {
-                return Content((HttpStatusCode)400, "Invalid SSO ID");
-            }
-            using (var _db = new DatabaseContext())
-            {
+
+                _db = new DatabaseContext();
+
                 List<Session> sessions = null;
                 User user = null;
-                try
+
+                var _userManagementManager = new UserManagementManager(_db);
+                user = _userManagementManager.GetUser(userSSOID);
+                if (user == null)
                 {
-                    var _userManagementManager = new UserManagementManager(_db);
-                    user = _userManagementManager.GetUser(userSSOID);
-                    if (user == null)
-                    {
-                        return Ok("User was never registered.");
-                    }
-                    var _sessionService = new SessionService();
-                    sessions = _sessionService.GetSessions(_db, userSSOID);
-                    if (sessions != null)
-                    {
-                        foreach (var sess in sessions)
-                        {
-                            _sessionService.DeleteSession(_db, sess.Token);
-                        }
-                    }
-                    UserManagementManager _userManager = new UserManagementManager(_db);
-                    _userManager.DeleteUser(userSSOID);
-                    _db.SaveChanges();
-                    return Ok("User was deleted");
+                    return Ok("User was never registered.");
                 }
-                catch (InvalidDbOperationException)
+                var _sessionService = new SessionService();
+                sessions = _sessionService.GetSessions(_db, userSSOID);
+                if (sessions != null)
                 {
-                    _db.Entry(user).State = System.Data.Entity.EntityState.Detached;
-                    if (sessions != null)
+                    foreach (var sess in sessions)
                     {
-                        _db.Entry(sessions).State = System.Data.Entity.EntityState.Detached;
+                        _sessionService.DeleteSession(_db, sess.Token);
```

Comment:
Provides additional state checks behind the scenes and does the same thing as the replaced line.